tracing-appender/src/non_blocking.rs

```diff
@@ -244,27 +245,25 @@ impl MakeWriter for NonBlocking {
 }
 
 impl WorkerGuard {
-    fn new(handle: JoinHandle<()>, shutdown_signal: Arc<AtomicBool>) -> Self {
+    fn new(handle: JoinHandle<()>, sender: Sender<Msg>) -> Self {
         WorkerGuard {
             guard: Some(handle),
-            shutdown_signal,
-        }
-    }
-
-    fn stop(&mut self) -> std::thread::Result<()> {
-        match self.guard.take() {
-            Some(handle) => handle.join(),
-            None => Ok(()),
+            sender,
         }
     }
 }
 
 impl Drop for WorkerGuard {
     fn drop(&mut self) {
-        self.shutdown_signal.store(true, Ordering::Relaxed);
-        match self.stop() {
+        match self
+            .sender
+            .send_timeout(Msg::Shutdown, Duration::from_millis(100))
+        {
             Ok(_) => (),
-            Err(e) => println!("Failed to join worker thread. Error: {:?}", e),
+            Err(e) => println!(
```

Comment:
We should probably check if the error here is that the receiver has disconnected, before we print anything. If it has, that indicates that the worker has already terminated, which is totally fine and we don't need to print anything.

Reply:
Good catch.